### PR TITLE
Upgrade the minimum compiler C++17

### PR DESCRIPTION
**Pull request by cajun-rat**
See discussion on https://github.com/uptane/aktualizr/issues/106

Full C++17 support came in gcc 8 (although most of it was present in gcc 7 already. In terms of our likely target platforms already ship that:

* Debian oldstable shipped with gcc-10.
* The oldest supported Yocto release is Dunfell, which shipped gcc 9.5

That suggests that getting a C++17 compiler shouldn't be a huge problem for our downstream.

Why not go further? C++20 support would require gcc 10 or 11, and while we could go that far it seems like we should be conservative by default.

### CODE REVIEW

**Comment by cajun-rat** [comment]
Grr, this triggers new clang-tidy checks

File: src/libaktualizr/crypto/p11engine.cc [hunk 1]
```diff
@@ -78,7 +78,7 @@ P11Engine::P11Engine(boost::filesystem::path module_path, std::string pass)
   LOG_DEBUG << "Slot token model.......: " << slot->token->model;
   LOG_DEBUG << "Slot token serialnr....: " << slot->token->serialnr;
 
-  uri_prefix_ = std::string("pkcs11:serial=") + slot->token->serialnr + ";pin-value=" + pass + ";id=%";
+  uri_prefix_ = std::string("pkcs11:serial=") + slot->token->serialnr + ";pin-value=" + pass_ + ";id=%";
```
Comment (rborn-tx) [inline src/libaktualizr/crypto/p11engine.cc:81]: That's a bug fix on its own, isn't it?

**Review by rborn-tx — APPROVED** [review]
LGTM

File: src/libaktualizr/crypto/p11engine.cc (hunk 1 above)
Comment (cajun-rat) [reply src/libaktualizr/crypto/p11engine.cc:81]: Yes. It was caught by the new compiler warnings :)

**Pull request merged**